### PR TITLE
update sdk and cli to add partner api invocation

### PR DESCRIPTION
**Pull request by khyberspache**

### CODE REVIEW

File: python/sdk/prelude_sdk/controllers/detect_controller.py [hunk 1]
```diff
@@ -187,3 +187,33 @@ def make_decision(self, id: str, decision: int):
...
+                return res.json()
+            raise Exception(res.text)
+
+    @verify_credentials
+    def partner_deploy(self, control_name: str, host_ids: list[str]):
+        """ Deploy probes on all specified partner endpoints """
+        with Spinner():
+            params = dict(host_ids=host_ids, control=control_name)
+            res = requests.post(
+                f'{self.account.hq}/iam/partner',
```
[the remark this answers is not in the document]
Comment (privateducky) [reply python/sdk/prelude_sdk/controllers/detect_controller.py:212]: in a new controllers/partner_controller.py

File: python/cli/prelude_cli/views/detect.py [hunk 2]
```diff
@@ -164,4 +164,24 @@ def describe_activity(controller, days, view, tests, tags, endpoints, dos):
...
+    print_json(controller.partner_endpoints(platform=platform, hostname=hostname, offset=offset))
+
+
+@detect.command('partner-deploy')
+@click.confirmation_option(prompt='Are you sure?')
+@click.option('--control', required=True, help='control name (e.g. "CrowdStrike")')
+@click.option('--host_ids', required=True, help='a list of host IDs to deploy to', type=list[str])
+@click.pass_obj
+@handle_api_error
+def detach_partner(controller, control, host_ids):
```
Comment (makk94) [inline python/cli/prelude_cli/views/detect.py:185]: also boop. name.

File: python/cli/prelude_cli/views/partner.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+@click.group()
+@click.pass_context
+def partner(ctx):
+    """ Continuous security testing """
+    ctx.obj = PartnerController(account=ctx.obj)
+
+
+@partner.command('endpoints')
+@click.option('--control_name', required=True, help='control name (e.g. "CrowdStrike")')
```
[the remark this answers is not in the document]
Comment (khyberspache) [reply python/cli/prelude_cli/views/partner.py]: done!

**Review by privateducky — APPROVED** [review]

**Comment by privateducky** [comment]
merge if happy

File: python/cli/prelude_cli/views/partner.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+@click.group()
+@click.pass_context
+def partner(ctx):
+    """ Continuous security testing """
+    ctx.obj = PartnerController(account=ctx.obj)
+
+
+@partner.command('endpoints')
+@click.option('--partner_name', required=True, help='control name (e.g. "CrowdStrike")')
```
Comment (mireaulf) [inline python/cli/prelude_cli/views/partner.py:17]: here too

**Review by mireaulf — APPROVED** [review]

File: python/cli/prelude_cli/views/partner.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    """ Get a list of endpoints from all partners """
+    print_json(controller.partner_endpoints(partner_name=partner_name, platform=platform, hostname=hostname, offset=offset))
+
+
+@partner.command('deploy')
+@click.confirmation_option(prompt='Are you sure?')
+@click.option('--partner_name', required=True, help='partner name (e.g. "CrowdStrike")')
+@click.option('--host_ids', required=True, help='a list of host IDs to deploy to', type=list[str])
+@click.pass_obj
+@handle_api_error
```
Comment (makk94) [inline python/cli/prelude_cli/views/partner.py:33]: partner_deploy

**Pull request merged**